br/pkg/storage/storage.go

```diff
@@ -216,3 +218,33 @@ func CloneDefaultHttpTransport() (*http.Transport, bool) {
 	transport, ok := http.DefaultTransport.(*http.Transport)
 	return transport.Clone(), ok
 }
+
+// GetMaxOffset returns the max offset of the file.
+func GetMaxOffset(ctx context.Context, storage ExternalStorage, name string) (n int, err error) {
```

Comment:
```suggestion
func GetMaxOffset(ctx context.Context, storage ExternalStorage, name string) (n int64, err error) {
```

so caller don't need type cast

Reply:
And the total file size has already been saved in s3ObjectReader.rangeInfo.Size, gcsObjectReader.totalSize, ... . I think we can extend the ExternalFileReader interface and return the value from them